app.js

```diff
@@ -9,14 +10,23 @@ app.get("/api/topics", getTopics);
 app.get("/api", (req,res,next) => {
     res.status(200).send({endpoints})
     next()
-   
 })
 
+app.get("/api/article/:article_id",getArticleById)
+
 app.all("/*", (req, res, next) => {
     res.status(404).send({ msg: "Path not found" });
   
 })
-
+app.use((err,req,res,next) =>{
+    
+    if (err.code === "22P02" && err.article_id){
+        res.status(400).send({message:`${err.article_id} is an invalid article_id (number)`})
+    }
+    else if (err.message === "Article not found"){
+        res.status(404).send({message:err.message})
+    }
```

Comment:
You can make this condition reusable for future errors coming from a Promise.reject rather than just if the message is 'article not found', it may make more sense when you come to another one for a future task though!